cpp/src/arrow/util/rle_encoding_internal.h

```diff
@@ -1079,7 +1084,6 @@ auto RleBitPackedDecoder<T>::GetSpaced(Converter converter,
   // There may be remaining null if they are not greedily filled by either decoder calls
   check_and_handle_fully_null_remaining();
 
-  ARROW_DCHECK(batch.is_done() || exhausted());
```

Comment:
This check could trigger if the RLE-bit-packed data is invalid (for example a run of invalid size). @AntoinePrv